lib/Site.js

```diff
@@ -357,24 +363,40 @@ Site.prototype.generatePages = function () {
   // Render the final rendered page to the output folder.
   const { baseUrl } = this.siteConfig;
   const builtFiles = {};
-  const pages = this.siteConfig.pages || [];
   const processingFiles = [];
+  const pages = this.filePaths || [];
   this.pageModels = pages.map(page => this.createPageData({
     baseUrl,
-    pageSrc: page.src,
-    title: page.title,
+    pageSrc: page,
     pageTemplate: this.pageTemplate,
+    titlePrefix: this.siteConfig.titlePrefix ? this.siteConfig.titlePrefix + TITLE_PREFIX_SEPARATOR : '',
   }));
   this.pageModels.forEach((page) => {
     processingFiles.push(page.generate(builtFiles));
   });
   return new Promise((resolve, reject) => {
     Promise.all(processingFiles)
-      .then(resolve)
+      .then((frontMatterData) => {
+        this.frontMatterData = frontMatterData;
+        resolve();
+      })
       .catch(reject);
   });
 };
 
+/**
+ * Builds the index of files to be traversed as addressable pages
+ */
+Site.prototype.buildIndex = function () {
```

Comment:
Rename this to `Site.prototype.collectAddressablePaths`.
Rename `this.filePaths` to `this.addressablePaths`.